dask_planner/src/expression.rs

```diff
@@ -146,14 +160,30 @@ impl PyExpr {
                         Ok(fq_name) => {
                             let mut idx: usize = 0;
                             for schema in input_plans[0].all_schemas() {
```

Comment:
I have a high level question about the logic here. It looks like we are iterating over multiple schemas but if we don't find a match on the first schema we will panic, so it looks like it can never look in more than one schema?

Reply:
indeed, that is not good. Let me adjust that

Reply:
I also don't think we should be using `all_schemas` here. I think we should just be looking at `input_plans[0].schema()`. I am not too familiar with `all_schema`, but for a join, it would return both the left scheme and the right schema for its inputs, so if we get an index into the right schema, that wouldn't make sense when applied to the output of the plan.

Reply:
So I'm not sure why, however when I remove `all_schema` to use `input_plans[0].schema()` a lot of test start failing with field not found errors from DataFusion? Could it be possible that maybe we do need `all_schemas()` like you that doesn't quite make sense to me but clearly something in there is needed.

Reply:
I'll make a commit with that change just so the changes and errors can be vizualized.

Reply:
I'm getting test failures locally before the commit that removes `all_schema`. Not sure if there is something up with my local env.